packages/opentelemetry-sdk-setup/src/sdk-setup.ts

```diff
@@ -0,0 +1,104 @@
+export const startTracing = async function (options: TracingOptions, packageJson: PackageJson) {
+  if (!options.preloadingEnabled) return
+  if (sdk) return
+
+  sdk = new HoneycombSDK({
+    resource: new Resource({
+      [SemanticResourceAttributes.SERVICE_VERSION]: packageJson.version,
+    }),
+    serviceName: packageJson.name,
+    protocol: 'grpc',
+    apiKey: options.apiKey,
+    endpoint: `${options.httpProtocol}://${options.host}:${options.port}`,
+    sampleRate: options.sampleRate,
+    // Turn off auto resource detection so that we fully control the attributes we export
+    autoDetectResources: false,
+  })
+
+  // Set the diagnostics logger to our system logger. We also need to suppress the override msg
+  // in case there's a default console logger already registered (it would log a msg to it)
+  diag.setLogger(getOtelLogger(options.debug, options.systemLogFile), {
+    logLevel: options.debug ? DiagLogLevel.DEBUG : DiagLogLevel.INFO,
+    suppressOverrideMessage: true,
+  })
+
+  sdk.start()
+
+  // Loads the contents of the passed baggageFilePath into the baggage
+  const baggageAttributes = await loadBaggageFromFile(options.baggageFilePath)
+  const baggageCtx = setMultiSpanAttributes(baggageAttributes)
+
+  const traceFlags = options.traceFlags !== undefined ? options.traceFlags : TraceFlags.NONE
+  // Sets the current trace ID and span ID based on the options received
+  // this is used as a way to propagate trace context from other processes such as Buildbot
+  if (options.traceId !== undefined && options.parentSpanId !== undefined) {
+    return trace.setSpanContext(baggageCtx, {
+      traceId: options.traceId,
+      spanId: options.parentSpanId,
+      traceFlags: traceFlags,
+      isRemote: true,
+    })
+  }
+
+  return context.active()
+}
+
+/** Stops the tracing service if there's one running. This will flush any ongoing events */
+export const stopTracing = async function () {
+  if (!sdk) return
+  try {
+    // The shutdown method might return an error if we fail to flush the traces
+    // We handle it and use our diagnostics logger
+    await sdk.shutdown()
+    sdk = undefined
+  } catch (e) {
+    diag.error(e)
+  }
+}
+
+/** Sets attributes to be propagated across child spans under the current active context
+ * TODO this method will be removed from this package once we move it to a dedicated one to be shared between build,
+ * this setup and any other node module which might use our open telemetry setup
+ * */
+export const setMultiSpanAttributes = function (attributes: { [key: string]: string }) {
+  const currentBaggage = propagation.getBaggage(context.active())
+  // Create a baggage if there's none
+  let baggage = currentBaggage === undefined ? propagation.createBaggage() : currentBaggage
+  Object.entries(attributes).forEach(([key, value]) => {
+    baggage = baggage.setEntry(key, { value })
+  })
+
+  return propagation.setBaggage(context.active(), baggage)
+}
```

Comment:
This is a copy/paste from:
- https://github.com/netlify/build/blob/97fd98c98c1a6d0ecbca8a5e5ca14867d3e6053e/packages/build/src/tracing/main.ts#L81-L91

Which we currently need in order to setup the context passed over to this process (baggage and trace information) to do the tracing and attributes stitching. Once we move the remainder of logic in build's tracing directory to its own utility library we can remove this and use that directly.